packages/mobx-react-lite/src/useObserver.ts

```diff
@@ -5,6 +5,9 @@ import { isUsingStaticRendering } from "./staticRendering"
 import { observerFinalizationRegistry } from "./utils/observerFinalizationRegistry"
 import { useSyncExternalStore } from "use-sync-external-store/shim"
 
+// Required by SSR when hydrating #3669
+const getServerSnapshot = () => {}
```

Comment:
I doubt that this is the intended snapshot, but no idea what else to put here, serializability is an assumption we cannot make true